optaweb-employee-rostering-frontend/src/ui/components/WeekPicker.tsx

```diff
@@ -90,20 +118,19 @@ export default class WeekPicker extends React.Component<WeekPickerProps, WeekPic
           </svg>
         </Button>
         <DatePicker
-          className="week-picker"
-          locale={
-            /* moment intreprets "en" as "en-US", this intreprets "en" as "en-GB" */
-            locale === 'en' ? 'en-US' : locale
-          }
-          value={[getFirstDayInWeek(this.props.value), getLastDayInWeek(this.props.value)]}
-          onChange={() => this.goToCurrentWeek()}
-          onClickDay={(value: Date) => this.goToWeekContaining(value)}
-          onCalendarOpen={() => this.setState({ isOpen: true })}
-          onCalendarClose={() => this.setState({ isOpen: false })}
+          selected={getFirstDayInWeek(this.props.value)}
+          startDate={getFirstDayInWeek(this.props.value)}
+          endDate={getLastDayInWeek(this.props.value)}
+          onChange={(date) => {
+            if (date !== null) {
+              this.goToWeekContaining(date);
+            } else {
+              this.goToCurrentWeek();
+            }
+          }}
           minDate={this.props.minDate}
           maxDate={this.props.maxDate}
-          isOpen={this.state.isOpen}
-          clearIcon={<HistoryIcon />}
+          customInput={<DateRangeInputElement />}
```

Comment:
```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of `r`.
    in DateRangeInputElement (at WeekPicker.tsx:140)
    in div (created by r)
    in div (created by InnerReference)
    in InnerReference (created by Context.Consumer)
    in Reference (created by r)
    in Manager (created by r)
    in r (created by r)
    in r (at WeekPicker.tsx:126)
    in div (at WeekPicker.tsx:100)
    in WeekPicker (at AvailabilityRosterPage.tsx:393)
    in span (at AvailabilityRosterPage.tsx:370)
    in AvailabilityRosterPage (created by Context.Consumer)
    in withRouter(AvailabilityRosterPage) (created by ConnectFunction)
    in ConnectFunction (created by withI18nextTranslation(Connect(withRouter(AvailabilityRosterPage))))
    in withI18nextTranslation(Connect(withRouter(AvailabilityRosterPage))) (created by Context.Consumer)
    in Route (at App.tsx:86)
    in Switch (at App.tsx:55)
    in section (created by PageSection)
    in PageSection (at App.tsx:45)
    in main (created by Page)
    in div (created by Page)
    in Page (at App.tsx:39)
    in App (at src/index.tsx:59)
    in Suspense (at src/index.tsx:58)
    in Router (created by BrowserRouter)
    in BrowserRouter (at src/index.tsx:57)
    in Provider (at src/index.tsx:56)
    in I18nextProvider (at src/index.tsx:55)
```
If I understand it correctly, `DateRangeInputElement` is a function component but the render method of `DatePicker` tries to pass a ref to it, which is kinda not allowed. This looks fishy. Can you try fixing this?

A similar situation is described and solved in https://stackoverflow.com/questions/58574298/warning-with-function-components-cannot-be-given-refs-even-though-using-forwar.

Reply:
Their custom input example does not use useForwardRef: https://reactdatepicker.com/. However, I can confirm using forwardRef fixes the error.

Reply:
Perfect, thanks!